modules/api/src/main/java/org/apache/ignite/sql/Criteria.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.sql;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Represents a predicate. Implementations of this interface are basic building blocks for performing criteria queries.
+ */
+public interface Criteria {
+    /**
+     * Create a predicate for testing the column is equal to a given value.
+     *
+     * @param columnName Column name.
+     * @param value Column value.
+     * @return the created <b>equal</b> predicate instance.
+     */
+    public static <R> Criteria equal(String columnName, Comparable<R> value) {
```

Comment:
```suggestion
    public static <R> Criteria equal(String columnName, Object value) {
```
`byte[]` is valid argument for binary column, but arrays are not Comparable in Java.